### ISSUE TITLE
Various Requirement Checks

### ISSUE THREAD

**Issue by AllenSeitz** [Dessyreqt/alttprandomizer#20]
The top right tombstone in the graveyard (where you get the Magic Cape) is listed as having a requirement of Boots && Titans to access. The Boots are required to bonk the stairs open, but the Titans are not strictly required. If you can reach the lower Dark World then the Mirror can also get you around the dark rocks.

The 3 chests in the castle escape basement (C-B1) are listed as having a requirement of Boots && Glove. Only the Glove is needed. Unlike the tombstone that leads to the Cape, this one can be simply pushed. And the walls can be bombed.

You listed the requirement for entering Hyrule Castle Tower as "Master Sword || Cape". Which was news to me, but I believe it. However I don't think using the Cape there is fair in casual difficulty. Also your requirements for CanDefeatAgahnim1() are the same. And that is not true, especially if you use the Cape to sneak in. CanDefeatAgahnim1() should probably check for "Master Sword || Net". (Or Shovel, but that's not fair because the Ocarina replaces it.) And then killing Agahnim with the Net isn't casual either.

CanDefeatSwampPalace() only checks for the Hookshot. But the level 1 sword does not damage the cotton balls surrounding Arrghus. Technically one of many other things is required too. (This could only matter if you start shuffling the bosses or something.)

**Comment by Dessyreqt** [comment]
Your first two points are valid and have been addressed in the code.

In randomizer (and perhaps in JP v1.0 as well since I don't think we changed this,) the L1 sword can reflect shots back to Agahnim. As far as this belonging in Casual difficulty... sneaking into Hyrule Castle Tower with Cape is something I have known for at least 20 years (The DMD has been known for at least that long) so maybe I'm biased, but Randomizer is meant to be slightly more difficult than the normal game, and knowledge will always gain you an advantage in randomizer. I'm electing not to change this.

Finally I went and tested Arrghus using only hookshot and the L1 sword. I found I had no difficulty defeating it using the secret technique handed down by our people (the spin attack.) There will also be no change here.
